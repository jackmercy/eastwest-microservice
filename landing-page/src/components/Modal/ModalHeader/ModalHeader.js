import React, { memo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import useStyles from './modalHeader.styles';

const ModalHeader = ({ title, children, className, toggle, ...restProps }) => {
  const classes = useStyles();

  return (
    <div className={classNames(classes.root, className)} {...restProps}>
      {toggle ? (
        <span
          role="presentation"
          onClick={toggle}
          className={classes.closeButton}
          data-test="component-modal-header-button-close"
        >
          <i className="icon-close" />
        </span>
      ) : null}
      <h2 className={classes.title} data-test="component-modal-header-title">
        {title}
      </h2>
      {children}
    </div>
  );
};
ModalHeader.propTypes = {
  title: PropTypes.node.isRequired,
  toggle: PropTypes.func,
  children: PropTypes.node,
  className: PropTypes.string,
};
ModalHeader.defaultProps = {
  className: '',
  toggle: undefined,
  children: null,
};

export default memo(ModalHeader);
