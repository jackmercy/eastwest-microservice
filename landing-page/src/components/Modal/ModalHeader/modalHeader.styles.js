import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import COLORS from '../../../constants/COLORS';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    flex: 'none',
    position: 'relative',
    backgroundColor: 'inherit',
    padding: rem(32),
    transition: 'padding .5s ease',
    borderRadius: [rem(8), rem(8), 0, 0],
    borderBottom: `1px solid ${COLORS.borderGray01}`,
  },
  title: {
    fontWeight: 600,
    fontSize: rem(32),
    lineHeight: rem(32),
    textAlign: 'center',
    transition: 'font-size .5s ease',
    margin: 0,
  },
  closeButton: {
    position: 'absolute',
    top: rem(16),
    right: rem(16),
    fontSize: rem(16),
    display: 'flex',
    cursor: 'pointer',
    transition: 'font-size .5s ease',
    '&:hover': { color: COLORS.primary },
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    root: { padding: rem(16) },
    title: { fontSize: rem(16), lineHeight: rem(24), marginTop: rem(8) },
  },
});
