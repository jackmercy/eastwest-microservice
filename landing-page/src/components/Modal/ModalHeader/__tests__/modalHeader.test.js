import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ModalHeader from '../ModalHeader';

describe('Testing Modal component > ModalHeader', () => {
  let wrapper;
  const props = { title: 'Modal header', toggle: jest.fn() };

  beforeAll(() => {
    wrapper = mount(<ModalHeader {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<ModalHeader {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a title', () => {
    expect(
      wrapper.find('[data-test="component-modal-header-title"]'),
    ).toHaveLength(1);
  });

  it('should have a button close', () => {
    expect(
      wrapper.find('[data-test="component-modal-header-button-close"]'),
    ).toHaveLength(1);
  });
});
