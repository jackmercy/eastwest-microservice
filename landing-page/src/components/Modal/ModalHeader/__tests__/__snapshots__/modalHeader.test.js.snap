// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Modal component > ModalHeader should render match snapshot 1`] = `
<div
  className="root-0-2-1"
>
  <span
    className="closeButton-0-2-3"
    data-test="component-modal-header-button-close"
    onClick={[MockFunction]}
    role="presentation"
  >
    <i
      className="icon-close"
    />
  </span>
  <h2
    className="title-0-2-2"
    data-test="component-modal-header-title"
  >
    Modal header
  </h2>
</div>
`;
